modules/integration/src/test/scala/scala/cli/integration/BspTestDefinitions.scala

```diff
@@ -851,6 +851,41 @@ abstract class BspTestDefinitions extends ScalaCliSuite with TestScalaVersionArg
           val resp = await(remoteServer.buildTargetCompile(new b.CompileParams(targets)).asScala)
           expect(resp.getStatusCode == b.StatusCode.OK)
         }
+
+        expect(changeFuture.isCompleted)
```

Comment:
I don't think we're awaiting this future anywhere, shouldn't we? otherwise there could be a race here.

Reply:
This promise is fulfilled on receiving `DidChangeBuildTarget`, this message should arrive when we trigger compilation with `remoteServer.buildTargetCompile`. See `BspImpl.scala` line 296 - `buildChanged` flag is set in the result of `prepareBuild()` if bloop config json file is changed for a build target.